src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtilTest.java

```diff
@@ -75,6 +92,44 @@ public void testMetadataFilesGenerationAllFiles() throws Exception {
                         + "module", checkstyleModules, metaFiles);
     }
 
+    @Test
+    public void testMetadataFilesWithNoDescription() throws Exception {
+
+        final String path = System.getProperty("user.dir")
+                + "/src/main/java/com/puppycrawl/tools/checkstyle";
+
+        final DefaultConfiguration scraperCheckConfig =
+                createModuleConfig(JavadocMetadataScraper.class);
+        final DefaultConfiguration treeWalkerConfig =
+                createModuleConfig(TreeWalker.class);
+
+        treeWalkerConfig.addChild(scraperCheckConfig);
+
+        final Checker checker = createChecker(treeWalkerConfig);
+
+        final List<File> validFiles = MetadataGeneratorUtil.extractValidFiles(path);
```

Comment:
Like I said, it makes more sense generate be returning all this. Generate is what we will be calling with in other repos, not the extract. Can we hook it all into 1 method?

I am having a hard time discerning the difference between `JavadocMetadataScraperTest` and `MetadataGeneratorUtilTest`. I expected `JavadocMetadataScraperTest` to do the type of validation you wrote here, and `MetadataGeneratorUtilTest` to do unit tests for the basic functionality of the utility.